doc/man3/OSSL_HPKE_CTX_new.pod

```diff
@@ -84,48 +84,159 @@ is likely required before using these APIs.  HPKE is used by various
 other IETF specifications, including the (draft) TLS Encrypted Client 
 Hello (ECH) specification and others.
 
-HPKE is a highly flexible consstruct requiring supporting many optional
-inputs for the various combinations in which modes and algorithms can
-be used.
-
-Notes on the various operations supported:
-
-- OSSL_HPKE_seal() is used by senders to encrypt a value to a
-recipient (supplying optional additional authenticated data or
-AAD)
+HPKE is a highly flexible consstruct supporting many optional inputs for the
+various combinations in which modes and algorithms can be used.
+
+=head2 Define'd symbols and data strutures
+
+- Symbols are defined (in include/openssl/hpke.h) for each of the
+IANA-allocated codepoints for HPKE modes ,e.g. OSSL_HPKE_MODE_BASE,
+key encapsulation methods (KEMs), e.g. OSSL_HPKE_KEM_ID_X25519, key
+derivation functions (KDFs), e.g. OSSL_HPKE_KDF_ID_SHA256 and
+authenticated encryption with additional data (AEAD) algorithms,
+e.g. OSSL_HPKE_AEAD_ID_AES_GCM_128.
+
+- The combination of a KEM, KDF and AEAD is represented in the
+OSSL_HPKE_SUITE data structure which maps to how HPKE is used
+in various protocols, and in these APIs.
+
+    typedef struct {
+        uint16_t    kem_id;
+        uint16_t    kdf_id;
+        uint16_t    aead_id;
+    } OSSL_HPKE_SUITE;
+
+- String values for the names of modes and suites are also
+defined, to assist with configuration settings, and an API is
+also provided (OSSL_HPKE_str2suite()) for mapping from those
+strings to an OSSL_HPKE_SUITE data structure.
+
+- An OSSL_HPKE_CTX is an opaque data structure that maintains
+internal state as HPKE operations are carried out.
+
+=head2 Context Construct/Free
+
+- OSSL_HPKE_CTX_new() sets up an opaque instance for internal HPKE
+state as HPKE operations are carried out. The inputs required are
+the HPKE mode and suite. The caller can optionally supply an overall library
+context (see L<OSSL_LIB_CTX(3)>, and property query I<propq>.
+
+- OSSL_HPKE_CTX_free() releases memory associated with
+an OSSL_HPKE_CTX pointer, including free'ing the top level
+data structure.
+
+Each OSSL_HPKE_CTX can be used for multiple calles to OSSL_HPKE_seal()
+or OSSL_HPKE_open(), but only a single call to OSSL_HPKE_encap() or
+OSSL_HPKE_decap() can be done with a given OSSL_HPKE_CTX pointer.
+In other words, calling OSSL_HPKE_encap() or OSSL_HPKE_decap() twice with the same
+OSSL_HPKE_CTX will result in an error. It is also an error to use
+the same OSSL_HPKE_CTX for both sender and receiver, so if e.g. one
+process acts as both sender and reciver then two instances of 
+OSSL_HPKE_CTX will be needed.
+
+=head2 Sender APIs
```

Comment:
Have another look at some other pod files to see how they are layed out and how the each parameter is documented with I<paramname>.

You could still have a section related to senders and recipients but I would do it separately when you just describe things in general terms..